core/src/main/java/com/microsoft/applicationinsights/common/SanitizationUtils.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.applicationinsights.common;
+
+import java.text.StringCharacterIterator;
+
+/**
+ * Created by dhdoshi on 09/10/2017
+ *
+ * This class provides utility functions to sanitize strings
+ * for various formats. Currently it supports JSON sanitization
+ */
+public final class SanitizationUtils {
```

Comment:
why you want it to be public?

Reply:
@SergeyKanzhelev I think we can't make a class private and I believed it would be best to create a class for sanitization and put all related methods there. In the future we might need to sanitize for eg URL or something else we can put this there. It also helps in better organization for testing purpose too.

Reply:
why you cannot make class internal? What' the idea of exposing this API publicly?

Reply:
@SergeyKanzhelev I don't purposefully intend to make an API public. I totally agree that it can added unnecessary constraint of maintaining it as people can take dependency over it. In my previous comment I did not meant that we can't create inner classes in Java. I can well make this class as an inner class inside JsonTelemetryDataSerializer / even down just create a private method there itself. My thoughts were that it might not be the best place as this method may have utility in other parts of the SDK (though not sure) and in that case we might not have the leverage to reuse the same method. Further, putting it inside JsonTelemetryDataSerializer might make it difficult to test as its a final class and doesn't implement and interface. That means we can't mock the behavior of it with Mockito framework and hence the only option would be to create the actual object of the larger class with more external dependencies just for the mere purpose of testing. PS: this are just my thoughts from very limited experience and I can be totally wrong. Please provide your guidance on what you think would be the best for this scenario.

Reply:
You can make it package-private by removing the `public` keyword. This way you will have minimal code change and easy to read code without exposing these helpers to the world.